diff --git a/node_modules/framework7/components/tooltip/tooltip.js b/node_modules/framework7/components/tooltip/tooltip.js
index c4ffd27..533b1cc 100644
--- a/node_modules/framework7/components/tooltip/tooltip.js
+++ b/node_modules/framework7/components/tooltip/tooltip.js
@@ -99,13 +99,12 @@ export default {
     },
 
     pageBeforeRemove(page) {
-      const app = this;
-      page.$el.find('.tooltip-init').each(el => {
+      page.$el.find('*').each(el => {
         if (el.f7Tooltip) el.f7Tooltip.destroy();
       });
 
-      if (app.theme === 'ios' && page.view && page.view.router.dynamicNavbar && page.$navbarEl && page.$navbarEl.length > 0) {
-        page.$navbarEl.find('.tooltip-init').each(el => {
+      if (page.$navbarEl && page.$navbarEl.length > 0) {
+        page.$navbarEl.find('*').each(el => {
           if (el.f7Tooltip) el.f7Tooltip.destroy();
         });
       }
diff --git a/node_modules/framework7/modules/router/router-class.js b/node_modules/framework7/modules/router/router-class.js
index 01963aa..c35bb06 100644
--- a/node_modules/framework7/modules/router/router-class.js
+++ b/node_modules/framework7/modules/router/router-class.js
@@ -1072,6 +1072,10 @@ class Router extends Framework7Class {
     $pageEl.trigger(colonName, page);
     router.emit(camelName, page);
 
+    if (callback === 'afterIn' && page.pageFrom) {
+      page.pageFrom = null;
+    }
+
     if (callback === 'beforeRemove' || callback === 'beforeUnmount') {
       detachEvents();
 
diff --git a/node_modules/framework7/shared/utils.d.ts b/node_modules/framework7/shared/utils.d.ts
index c88ac4c..e29088b 100644
--- a/node_modules/framework7/shared/utils.d.ts
+++ b/node_modules/framework7/shared/utils.d.ts
@@ -66,7 +66,7 @@ export interface Utils {
   /** Returns unique number, increased by 1 with every call */
   uniqueNumber: () => number;
   /** Generates random ID-like string */
-  id: (mask: string, map: string) => string;
+  id: (mask?: string, map?: string) => string;
   /** Returns preloader inner content for MD theme */
   mdPreloaderContent: () => string;
   /** Returns preloader inner content for iOS theme */
